components/base-node.js

```diff
@@ -65,7 +79,32 @@ module.exports = {
        ar.push(item);
      }
      return ar;
-   } 
+   },
+
+   updatePromise: function(def) {
+     return promiseComponent(
 
+         _.defaults(
+             def,
+             { inPorts: _.mapObject(
+                          def.inPorts,
+                          function(port, portName) {
+
+                            return _.defaults(
+                                     port,
+                                     {
+                                       ondata: function ( payload, socketIndex ) {
+                                                 this[portName] = payload;
```

Comment:
This will be changing.  I actually removed some of my latest vni code from this API in an attempt to keep this review focused on the json_splitter component without getting into the vni manager that will be called from here.

The updatePromise API is used to extend the generic promiseComponent to call the component's onchange API.  Without it, you wouldn't get the custom component data handling. 

At the time ondata function is executed, we will be running on the node context.